src/test/Utils.sol

```diff
@@ -1,67 +1,210 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity ^0.8.10;
 
-import {Hevm} from "solmate/test/utils/Hevm.sol";
+import {DSTest} from "ds-test/src/test.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {Vm} from "forge-std/src/Vm.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+import {AludelFactory} from "../contracts/AludelFactory.sol";
+
+import {IAludel} from "../contracts/aludel/IAludel.sol";
+
+import {MockERC20} from "../contracts/mocks/MockERC20.sol";
+import {User} from "./User.sol";
+
+import {Crucible, IUniversalVault} from "alchemist/contracts/crucible/Crucible.sol";
+import {CrucibleFactory} from "alchemist/contracts/crucible/CrucibleFactory.sol";
 
 library Utils {
-    function getCheatcodes() public returns (Hevm) {
-        return Hevm(address(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D));
+    struct LaunchParams {
+        address template;
+        string name;
+        string stakingTokenUrl;
+        uint64 startTime;
+        address vaultFactory;
+        address[] bonusTokens;
+        address owner;
+        bytes initParams;
+    }
+
+    string public constant LOCK_EVENT = "Lock";
+    string public constant UNLOCK_EVENT = "Unlock";
+
+    function vm() public returns (Vm) {
+        return Vm(address(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D));
+    }
+
+    function launchProgram(
+        AludelFactory factory,
+        LaunchParams memory params
+    ) internal returns(address program) {
+        program = factory.launch(
+        params.template,
+            params.name,
+            params.stakingTokenUrl,
+            params.startTime,
+            params.vaultFactory,
+            params.bonusTokens,
+            params.owner,
+            params.initParams
+        );
+    }
+
+    function createInstance(address owner, CrucibleFactory crucibleFactory) internal returns (Crucible crucible) {
+        vm().prank(owner);
+        return Crucible(payable(crucibleFactory.create("")));
+    }
+
+    function createInstance(User user, CrucibleFactory crucibleFactory) internal returns (Crucible crucible) {
```

Comment:
It's not obvious to me an instance _of what_ it's creating. I'd advise to rename it to createCrucible or something like that

Reply:
This comes from the old alchemist hardhat tasks, `create-instance` is the task used to create crucible instance. I've updated the name to `createCrucible`.